File: src/main/java/org/tensorflow/data/GraphLoader.java
package org.tensorflow.data;

import org.tensorflow.Operand;
import org.tensorflow.Session;
import org.tensorflow.op.Ops;
import org.tensorflow.types.family.TType;

public interface GraphLoader<T extends TType> extends Dataset<T> {

  Operand<T>[] getBatchOperands();

  void build(Ops tf);

  long size();

  Session.Runner feedSessionRunner(Session.Runner runner, long batch);
}

File: src/main/scala/org/tensorflow/data/GraphLoader.scala
package org.tensorflow.data

import org.tensorflow.Operand
import org.tensorflow.Session
import org.tensorflow.op.Ops
import org.tensorflow.types.family.TType

trait GraphLoader[T <: TType] extends Dataset[T] {
  def getBatchOperands: Array[Operand[T]]

  def build(tf: Ops): Unit

  def size: Long

  def feedSessionRunner(runner: Session#Runner, batch: Long): Session#Runner
}
